server/repository/src/migrations/v2_06_00/add_cancellation_fields_to_invoice.rs

```diff
@@ -0,0 +1,26 @@
+use crate::migrations::*;
+pub(crate) struct Migrate;
+
+impl MigrationFragment for Migrate {
+    fn identifier(&self) -> &'static str {
+        "add_cancellation_fields_to_invoice"
+    }
+
+    fn migrate(&self, connection: &StorageConnection) -> anyhow::Result<()> {
+        let result = sql!(
+            connection,
+            r#"          
+                ALTER TABLE invoice
+                ADD COLUMN is_cancellation BOOLEAN NOT NULL DEFAULT FALSE;
+                ALTER TABLE invoice
+                ADD COLUMN cancelled_datetime {DATETIME};
+            "#,
+        );
+
+        if result.is_err() {
```

Comment:
Prevents panic if the fields already exist (for people who ran the updated (and now reverted) version of the migration)